client/js/player/movement.js

```diff
@@ -62,23 +66,21 @@ function simulate(collides, state, c, dt) /* newState */ {
 	rt *= 10 * dt;
 	var delta = {
 		x: -cos(c.lon)*fw + sin(c.lon)*rt,
-		y: newState.vy * dt,
-		z: -sin(c.lon)*fw - cos(c.lon)*rt,
+		y: newState.entityState.body.vel.y * dt,
+		z: -sin(c.lon)*fw - cos(c.lon)*rt
 	};
 
-	newState.pos.copy(state.pos);
-
-	delta = attemptMove(newState.pos, delta, collides);
+	delta = attemptMove(newState.entityState.body, delta, collides);
 
 	if (delta.y === 0) {
-		newState.vy = c.jump ? 6 : 0;
+		newState.entityState.body.vel.y = c.jump ? 6 : 0;
```

Comment:
Like, why not just pass newState.entityState.body into this function, if it doesn't use any of the other data?